config/peerpods/podvm/libvirt-podvm-image-handler.sh

```diff
@@ -0,0 +1,273 @@
+    [[ -z "${CAA_REF}" ]] && error_exit "CAA_REF is empty"
+
+    # Ensure booleans are set
+    [[ -z "${DOWNLOAD_SOURCES}" ]] && error_exit "DOWNLOAD_SOURCES is empty"
+}
+
+
+
+function create_libvirt_image() {
+    echo "Creating libvirt qcow2 image"
+
+    # If any error occurs, exit the script with an error message
+
+    if [[ "${DOWNLOAD_SOURCES}" == "yes" ]]; then
+        # Download source code from GitHub
+        download_source_code
+    fi
+
+    # Prepare the source code for building the ami
+    prepare_source_code
+
+    #export PODVM_DISTRO=${PODVM_DISTRO}
+    #export CLOUD_PROVIDER=${CLOUD_PROVIDER}
+    #export ARCH=${ARCH}
+
+    cd "${CAA_SRC_DIR}"/podvm || \
+	    error_exit "Failed to change directory to ${CAA_SRC_DIR}/podvm"
+    make image
+
+    export PODVM_IMAGE_PATH=/payload/podvm-libvirt.qcow2
+    cp -pr ${CAA_SRC_DIR}/podvm/output/*.qcow2 $PODVM_IMAGE_PATH
+
+    # Upload the created qcow2 to the volume
+    upload_libvirt_image
+
+    # Get the libvirt pool uuid
+    # This will set the LIBVIRT_POOL_UUID environment variable
+    get_libvirt_pool_uuid
+
+    # Add the ami id as annotation to peer-pods-cm configmap
+    add_libvirt_uuid_annotation_to_peer_pods_cm
+}
+
+# Function to upload the qcow2 image to volume
+
+function upload_libvirt_image() {
+    echo "LIBVIRT_VOL_NAME: $LIBVIRT_VOL_NAME" && echo "LIBVIRT_POOL: $LIBVIRT_POOL" && \
+	    echo "LIBVIRT_URI: $LIBVIRT_URI" && echo "PODVM_IMAGE_PATH: $PODVM_IMAGE_PATH" 
+    echo "Starting to upload the image."
+    virsh -d 0 -c $LIBVIRT_URI vol-upload --vol $LIBVIRT_VOL_NAME $PODVM_IMAGE_PATH --pool $LIBVIRT_POOL --sparse
+    if [ $? -eq 0 ]; then
+        echo "Uploaded the image successfully"
+    fi
+}
+
+# Function to get the libvirt pood uuid of the newly created image
+
+function get_libvirt_pool_uuid() {
+    echo "Getting the libvirt pool uuid"
+
+    # Get the libvirt uuid of the uploaded qcow2 image
+    # If any error occurs, exit the script with an error message
+
+    LIBVIRT_POOL_UUID=$(virsh -d 0 -c $LIBVIRT_URI pool-list --name --uuid | grep $LIBVIRT_POOL | awk '{print $1}') ||
+        error_exit "Failed to get the libvirt uuid"
+
+    # Set the libvirt uuid as an environment variable
+    export LIBVIRT_POOL_UUID
+
+    echo "UUID of the libvirt pool: ${LIBVIRT_POOL_UUID}"
+}
+
+# Function to add the libvirt pool uuid as annotation in the peer-pods-cm configmap
+
+function add_libvirt_uuid_annotation_to_peer_pods_cm(){
+    echo "Adding libvirt uuid to peer-pods-cm configmap"
+
+    # Check if the peer-pods-cm configmap exists
+    if ! kubectl get configmap peer-pods-cm -n openshift-sandboxed-containers-operator >/dev/null 2>&1; then
+        echo "peer-pods-cm configmap does not exist. Skipping adding the libvirt uuid"
+        return
+    fi
+
+    # Add the libvirt pool uuid as annotation to peer-pods-cm configmap
+    kubectl patch configmap peer-pods-cm -n openshift-sandboxed-containers-operator \
```

Comment:
@snir911 awa/azure are adding as the annotation : https://github.com/openshift/sandboxed-containers-operator/blob/b13fc7111e920795df3a6f7a9e6e734cfa2539a9/config/peerpods/podvm/aws-podvm-image-handler.sh#L263 
But getImageID method looks for Data in configmap. Not sure how it is works.
https://github.com/openshift/sandboxed-containers-operator/blob/b13fc7111e920795df3a6f7a9e6e734cfa2539a9/controllers/image_generator.go#L556
I have patched as Data field in configmap.

Reply:
Right, in Azure and AWS the IMAGE_ID in the CM is used in order to identify the created podvm image, however, in libvirt, the equivalent is the volume name, which by default is "podvm-base.qcow2", unless LIBVIRT_VOL_NAME is set.
Actually i think you should drop LIBVIRT_POOL_UUID completely in favor of LIBVIRT_VOL_NAME, as the uuid is not consumed by CAA. 
cc @bpradipt for advise

Reply:
I agree.. if LIBVIRT_POOL_UUID is not used, then there is no point in adding.
If this will be required later to support per pod VM image, it can be added later

Reply:
We need one Libvirt related ID( CMimageIDKey)  LIBVIRT_POOL_UUID or LIBVIRT_VOL_NAME. This check is  required whether to  skip or perform the image configuration operation. 

We thought to use LIBVIRT_POOL_UUID_<Vol NAME> as a CMimageIDKey to differentiate between different Libvirt images. We planned to do corresponding changes in CAA also. 
This decided this way because two different LibVirt Pool can have same Volume names. This differentiation will be useful to support multiple PodVM images.

Now we will set CMimageIDKey = LIBVIRT_VOL_NAME for libVirt provider. In future, we will change as we said above.

Is it okay?

Reply:
yeah sounds good. Let's use LIBVIRT_VOL_NAME for now